### PR TITLE
[irods/irods#7247] Bumped Boost library to 1.81 (main)

### PR DESCRIPTION
**Pull request by korydraughn**
Will verify externals build for each OS shortly.

### CODE REVIEW

**Review by SwooshyCueb — APPROVED** [review]

File: versions.json [hunk 1]
```diff
@@ -3,7 +3,7 @@
         "commitish": "release-1.11.0",
         "version_string": "1.11.0",
         "license": "Apache License 2.0",
-        "consortium_build_number": "1",
+        "consortium_build_number": "0",
```
Comment (alanking) [inline versions.json:6]: Should this be a 2? See 0e87223b3f693bc26cdf5e7b146cce6f2a998529. Sorry if I missed this decision being made "offline"

Comment (korydraughn) [reply versions.json:6]: Good question.

@SwooshyCueb Thoughts? - When should the consortium_build_number be incremented vs reset?

Comment (korydraughn) [reply versions.json]: Bumped to 2.

**Review by alanking — APPROVED** [review]
Looks good to me. Deferring to @SwooshyCueb (and, of course, build working for our supported platforms)

File: versions.json (hunk 1 above)
Comment (trel) [reply versions.json]: if avro is not getting its own version bumped... and depends on boost... then moving to 2 is correct.

if avro is itself getting a bump, then it should be restarted at 0.

**Review by SwooshyCueb — APPROVED** [review]

**Comment by korydraughn** [comment]
All packages for the **server** target build without any issues for Ubuntu 20.

Getting ready to build the remaining packages for Ubuntu 20 now.

**Comment by korydraughn** [comment]
The build fails when it reaches mungefs.

**Comment by SwooshyCueb** [comment]
> The build fails when it reaches mungefs.

Unless mungefs uses boost, that's most likely a separate issue. I've not had any build issues with mungefs myself though. Which distros fail?

**Comment by korydraughn** [comment]
It failed for Ub20. I haven't looked into the problem just yet though.

**Comment by korydraughn** [comment]
It's failing because the CMakeLists.txt file for mungefs is pointing to the wrong version.

Will get a PR up for mungefs soon.

**Comment by korydraughn** [comment]
All externals built for Ub20.

Getting ready to compile for Debian 11.

File: versions.json [hunk 2]
```diff
@@ -237,7 +237,8 @@
         "fpm_directories": ["bin","include","lib"]
     },
     "mungefs": {
-        "commitish": "1.0.5",
+        "git_repository": "https://github.com/korydraughn/mungefs",
```
[the remark this answers is not in the document]
Comment (korydraughn) [reply versions.json:240]: Correct

**Comment by korydraughn** [comment]
Debian 11 packages built successfully.

File: versions.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (korydraughn) [reply versions.json:240]: @trel Please update our fork of boost.

**Comment by korydraughn** [comment]
CentOS 7 and Almalinux 8 packages built successfully.

**Comment by korydraughn** [comment]
Once our fork of boost is updated, I'll tweak the PR reference it and rebuilt externals for ub20 to confirm everything is good.

If all of that works, we'll be in a position to add the pound.

Oh, we need to remove Ubuntu 18 from the README as well.

**Comment by korydraughn** [comment]
I agree.

**Comment by SwooshyCueb** [comment]
Additionally, my understanding is that we still intend to release 4.3.1 on Ubuntu 18. It's not EOL until April 2028.

**Comment by korydraughn** [comment]
I'm seeing May 31, 2023.

**Comment by korydraughn** [comment]
Let's discuss tomorrow during standup.

File: versions.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (trel) [reply versions.json:240]: done.

**Comment by korydraughn** [comment]
Done.

**Comment by SwooshyCueb** [comment]
mungefs will need to point to the new sha or a new release, so that the resulting package will have the right dependencies.

**Comment by korydraughn** [comment]
Sure. I'm more so wondering if we can stop shipping mungefs as an external.

**Comment by alanking** [comment]
It is only used in testing. The tests which use it are all under the "timing" test suite, which I don't even think we run at this point. Maybe we *can* just drop it...

**Comment by korydraughn** [comment]
If it's only for testing, then I think it's safe to scrap the idea of version numbers for mungefs and simply use the SHA moving forward.

Thoughts?

**Comment by alanking** [comment]
Seems fine to me.

**Comment by SwooshyCueb** [comment]
That is a move away from normal and boring packaging.

**Comment by korydraughn** [comment]
Please elaborate.

**Comment by SwooshyCueb** [comment]
Sorry, I saw Alan's comment pop while I was typing, so I felt the need to make my objection known quickly.

I would very much prefer for things that we package to have proper version numbers. With more standard packaging (which we are moving towards), version strings get very complicated when not based on a specific release.

**Comment by korydraughn** [comment]
Okay. We'll bump the version number with the expectation that the situation will improve as we move into a more normal and boring packaging future.

@trel I think all that's needed is a version bump and 1.0.6 tag in the mungefs repo. Is that correct?

File: versions.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (korydraughn) [reply versions.json]: Removed git repo property.

**Comment by korydraughn** [comment]
This should be ready to go.

@alanking @SwooshyCueb Please take another look before I add the pound.

**Comment by korydraughn** [comment]
I also confirmed mungefs 1.0.6 is produced by this PR.

**Comment by trel** [comment]
seems good.  just version numbers are touched, and they align with what I expect.

**Review by alanking — APPROVED** [review]
Looks good to me. Please # once @SwooshyCueb approves

**Review by SwooshyCueb — APPROVED** [review]

**Comment by korydraughn** [comment]
#'d

**Pull request merged**